lib/puppet/http/session.rb

```diff
@@ -90,6 +90,9 @@ def process_response(response)
       site = Puppet::HTTP::Site.from_uri(response.url)
       @server_versions[site] = version
     end
+    if (compiler = response['X-Puppet-Compiler-Name'])
+      Puppet.notice("Catalog compiled by #{compiler}")
+    end
```

Comment:
I was thinking we'd only emit this in notice the `post_catalog` method: https://github.com/puppetlabs/puppet/blob/03b3df33deb2d792ffd1c0eed9721fdc1d4833ad/lib/puppet/http/service/compiler.rb#L115-L123

And possibly the `post_catalog4` method if it makes sense to do so? https://github.com/puppetlabs/puppet/blob/03b3df33deb2d792ffd1c0eed9721fdc1d4833ad/lib/puppet/http/service/compiler.rb#L159

Reply:
The change in puppetserver only affects the v3 endpoint, so the v4 POST is not necessary yet. It makes sense to me that we would just emit the notice anytime we received the header, so any endpoint the agent uses that the puppetserver deems necessary to include that header would "just work". For example, if we did start using the v4 catalog endpoint for agent runs we would just have to make changes on the server side and not the agent.

Is it possible that someone adds that header in a response that is not the puppetserver? It seems the same level of vulnerability as the `X-Puppet-Version` header. I'm not sure if `process_response` is used for non-puppetserver requests...